### DISCUSSION TITLE
Migrating Distance Metric from Euclid to Cosine

### DISCUSSION THREAD

**Discussion by gilljon** [qdrant/qdrant#1932] [Q&A]
Hey all,

I currently have my collections set up using `EUCLID` distance metric but I would like to convert them to use `COSINE`...

Is there a simple way to run migrations for my collections? I'd like to keep the existing points but just modify the distance metric.

Thanks 😄

**Comment by generall** [comment]
there is no automatic migration for this, unfortunately. Vector index structure depends on the metric type, so you would need to rebuild collection anyway
